src/compas/files/gltf/gltf_exporter.py

```diff
@@ -0,0 +1,396 @@
+            samplers_list[sampler_index_by_key[key]] = sampler_data.to_dict(input_accessor, output_accessor)
+        return samplers_list
+
+    def _set_initial_gltf_dict(self):
+        asset_dict = {'version': '2.0'}
+        gltf_dict = {'asset': asset_dict}
+        if self._content.extras:
+            gltf_dict['extras'] = self._content.extras
+        if self._content.extensions:
+            gltf_dict['extensions'] = self._content.extensions
+        self._gltf_dict = gltf_dict
+
+    def _add_scenes(self):
+        if not self._content.scenes:
+            return
+        if self._content.default_scene_key is not None:
+            self._gltf_dict['scene'] = self._scene_index_by_key[self._content.default_scene_key]
+        scene_list = [None] * len(self._content.scenes.values())
+        for key, scene in self._content.scenes.items():
+            scene_list[self._scene_index_by_key[key]] = scene.to_dict(self._node_index_by_key)
+        self._gltf_dict['scenes'] = scene_list
+
+    def _add_nodes(self):
+        if not self._content.nodes:
+            return
+        node_list = [None] * len(self._content.nodes)
+        for key, node in self._content.nodes.items():
+            node_list[self._node_index_by_key[key]] = node.to_dict(
+                self._node_index_by_key,
+                self._mesh_index_by_key,
+                self._camera_index_by_key,
+                self._skin_index_by_key,
+            )
+        self._gltf_dict['nodes'] = node_list
+
+    def _get_primitives(self, mesh_data):
+        primitives = []
+        for primitive_data in mesh_data.primitive_data_list:
+            indices_accessor = self._get_accessor(primitive_data.indices, COMPONENT_TYPE_UNSIGNED_INT, TYPE_SCALAR)
+
+            attributes = {}
+            for attr in primitive_data.attributes:
+                component_type = COMPONENT_TYPE_UNSIGNED_INT if attr.startswith('JOINT') else COMPONENT_TYPE_FLOAT
+                type_ = TYPE_VEC3
+                if len(primitive_data.attributes[attr][0]) == 4:
+                    type_ = TYPE_VEC4
+                if len(primitive_data.attributes[attr][0]) == 2:
+                    type_ = TYPE_VEC2
+                attributes[attr] = self._get_accessor(primitive_data.attributes[attr], component_type, type_, True)
+
+            targets = []
+            for target in primitive_data.targets or []:
+                target_dict = {}
+                for attr in target:
+                    component_type = COMPONENT_TYPE_FLOAT
+                    type_ = TYPE_VEC3
+                    target_dict[attr] = self._get_accessor(target[attr], component_type, type_, True)
+                targets.append(target_dict)
+
+            primitive_dict = primitive_data.to_dict(indices_accessor, attributes, targets, self._material_index_by_key)
+
+            primitives.append(primitive_dict)
+        return primitives
+
+    def _get_accessor(self, data, component_type, type_, include_bounds=False):
```

Comment:
These `_get_*` functions confuse me a bit. Not the content, but the name, because they do more than just getting stuff, they have side effects on the gltf dictionary. Would a prefix like `_allocate_` be more fitting than `_get_` perhaps?

Reply:
I went with `_construct_`.